topi/python/topi/x86/bitserial_conv2d.py

```diff
@@ -1,74 +1,33 @@
-from ..nn.bitserial_conv2d import _WORKLOADS, _SCH_TO_DECL_FUNC_QUANT
-
-_QUANTIZED_SCHEDULES_NCHW = [
-    # resnet
-    SpatialPackNCHW(2, 2, 8, 1, 1),
-    SpatialPackNCHW(1, 4, 8, 4, 1),
-    SpatialPackNCHW(1, 4, 8, 1, 16),
-    SpatialPackNCHW(1, 4, 8, 4, 8),
-    SpatialPackNCHW(1, 7, 8, 3, 8),
-    SpatialPackNCHW(1, 2, 8, 1, 8),
-    SpatialPackNCHW(2, 1, 8, 1, 4),
-    SpatialPackNCHW(1, 7, 8, 1, 1),
-    SpatialPackNCHW(1, 1, 8, 1, 16),
-    SpatialPackNCHW(1, 1, 8, 1, 8),
-    SpatialPackNCHW(1, 1, 8, 1, 16),
-
-    SpatialPackNCHW(3, 3, 16, 3, 16),
-    SpatialPackNCHW(1, 1, 16, 2, 16),
-    SpatialPackNCHW(1, 1, 8, 1, 16),
-    SpatialPackNCHW(1, 1, 8, 1, 16),
-]
-
-_QUANTIZED_SCHEDULES_NHWC = [
-    # resnet
-    SpatialPackNHWC(2, 2, 8, 1, 1),
-    SpatialPackNHWC(1, 4, 8, 4, 1),
-    SpatialPackNHWC(1, 4, 8, 1, 16),
-    SpatialPackNHWC(1, 4, 8, 4, 8),
-    SpatialPackNHWC(1, 7, 8, 3, 8),
-    SpatialPackNHWC(1, 2, 8, 1, 8),
-    SpatialPackNHWC(2, 1, 8, 1, 4),
-    SpatialPackNHWC(1, 7, 8, 1, 1),
-    SpatialPackNHWC(1, 1, 8, 1, 16),
-    SpatialPackNHWC(1, 1, 8, 1, 8),
-    SpatialPackNHWC(1, 1, 8, 1, 16),
-]
-
-@_get_schedule.register("cpu")
-def _get_schedule_bitserial_conv2d(wkl, layout):
-    if wkl not in _WORKLOADS:
-        raise ValueError("no schedule for such workload: {}".format(wkl))
-    idx = _WORKLOADS.index(wkl)
-    if layout == "NCHW":
-        sch = _QUANTIZED_SCHEDULES_NCHW[idx]
-    elif layout == "NHWC":
-        sch = _QUANTIZED_SCHEDULES_NHWC[idx]
-    return sch
-
-@bitserial_conv2d.register("cpu")
-def _declaration_bitserial_conv2d(data, kernel, stride, padding, activation_bits, weight_bits,
-                                  layout='NCHW', pack_dtype=None, out_dtype=None, dorefa=False):
-    if out_dtype is None:
-        out_dtype = data.dtype
-    assert data.shape[0].value == 1, "only support batch size=1 convolution on rasp"
-    assert layout in ("NCHW", "NHWC"), "only support layouts NCHW and NHWC"
-
-    wkl = _get_workload(data, kernel, stride, padding, out_dtype, layout)
-    sch = _get_schedule(wkl, layout)
-    return _SCH_TO_DECL_FUNC_QUANT[type(sch)](data, kernel, stride, padding, activation_bits,
-                                              weight_bits, pack_dtype, out_dtype, dorefa)
-
-@generic.schedule_bitserial_conv2d_nchw.register(["cpu"])
-@generic.schedule_bitserial_conv2d_nhwc.register(["cpu"])
-def schedule_bitserial_conv2d(outs):
+from ..nn.bitserial_conv2d import bitserial_conv2d_nhwc, bitserial_conv2d_nchw
+
+@autotvm.task.register("topi_x86_bitserial_conv_nhwc")
+def topi_bitserial_conv2d_nhwc(*args, **kwargs):
+    args = deserialize_args(args)
+    C = bitserial_conv2d_nhwc(*args, **kwargs)
+    s = generic.nn.schedule_bitserial_conv2d_nhwc([C])
+    data = args[0]
+    kernel = args[1]
+    return s, [data, kernel, C]
+
+@autotvm.task.register("topi_arm_cpu_bitserial_conv_nchw")
+@autotvm.task.register("topi_x86_bitserial_conv_nchw")
```

Comment:
what about moving task registration to `topi_integration` and renaming to `topi_bitserial_conv_nchw`?